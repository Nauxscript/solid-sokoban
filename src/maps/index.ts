export * from './level-1'

export const drawMap = ['', '🪨', '📦', '🎯', '🚶‍♂️', '💠']