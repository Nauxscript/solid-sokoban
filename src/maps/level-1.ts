import { MapData } from "./map.types";

export const level1:MapData = [
  [0, 0, 1, 1, 1, 0, 0, 0],
  [0, 0, 1, 3, 1, 0, 0, 0],
  [0, 0, 1, 0, 1, 1, 1, 1],
  [1, 1, 1, 2, 4, 2, 3, 1],
  [1, 3, 0, 2, 0, 1, 1, 1],
  [1, 1, 1, 1, 2, 1, 0, 0],
  [0, 0, 0, 1, 3, 1, 0, 0],
  [0, 0, 0, 1, 1, 1, 0, 0],
]