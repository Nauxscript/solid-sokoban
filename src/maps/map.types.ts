export type DotState = 0 | 1 | 2 | 3 | 4 

export type MapData = DotState[][]