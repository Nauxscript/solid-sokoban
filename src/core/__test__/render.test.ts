import { describe, expect, test } from 'vitest';
import { level1 } from '~/maps';
import { renderMap } from '../index';

describe('render test',() => {
  test('render map', () => {
    const renderedData = renderMap(level1, (dot: string) => dot)
    expect(renderedData).toMatchSnapshot()
  })
})