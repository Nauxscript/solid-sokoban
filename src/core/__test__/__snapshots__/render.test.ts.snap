// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render test > render map 1`] = `
[
  [
    "",
    "",
    "­Ъфе",
    "­Ъфе",
    "­Ъфе",
    "",
    "",
    "",
  ],
  [
    "",
    "",
    "­Ъфе",
    "­Ъј»",
    "­Ъфе",
    "",
    "",
    "",
  ],
  [
    "",
    "",
    "­Ъфе",
    "",
    "­Ъфе",
    "­Ъфе",
    "­Ъфе",
    "­Ъфе",
  ],
  [
    "­Ъфе",
    "­Ъфе",
    "­Ъфе",
    "­ЪЊд",
    "­ЪџХРђЇРЎѓ№ИЈ",
    "­ЪЊд",
    "­Ъј»",
    "­Ъфе",
  ],
  [
    "­Ъфе",
    "­Ъј»",
    "",
    "­ЪЊд",
    "",
    "­Ъфе",
    "­Ъфе",
    "­Ъфе",
  ],
  [
    "­Ъфе",
    "­Ъфе",
    "­Ъфе",
    "­Ъфе",
    "­ЪЊд",
    "­Ъфе",
    "",
    "",
  ],
  [
    "",
    "",
    "",
    "­Ъфе",
    "­Ъј»",
    "­Ъфе",
    "",
    "",
  ],
  [
    "",
    "",
    "",
    "­Ъфе",
    "­Ъфе",
    "­Ъфе",
    "",
    "",
  ],
]
`;
